src/portalbackend/PortalBackend.DBAccess/Repositories/CompanyCertificateRepository.cs

```diff
@@ -99,4 +99,39 @@ sorting switch
                 companyCertificate.ValidTill
                 ))
         .SingleOrDefaultAsync();
+
+    public Task<(Guid DocumentId, DocumentStatusId DocumentStatusId, IEnumerable<Guid> CompanyCertificateId, bool IsSameCompany)> GetCompanyCertificateDocumentDetailsForIdUntrackedAsync(Guid documentId, Guid companyId) =>
+            _context.Documents
+            .AsNoTracking()
+            .Where(x => x.Id == documentId)
+            .Select(document => new ValueTuple<Guid, DocumentStatusId, IEnumerable<Guid>, bool>(
+                    document.Id,
+                    document.DocumentStatusId,
+                    document.CompanyCertificates.Where(x => x.CompanyCertificateStatusId != CompanyCertificateStatusId.INACTVIE).Select(x => x.Id),
+                    document.CompanyUser!.Identity!.CompanyId == companyId))
+            .SingleOrDefaultAsync();
+
+    public void AttachAndModifyCompanyCertificateDetails(Guid id, Action<CompanyCertificate>? initialize, Action<CompanyCertificate> updateFields)
+    {
+        var entity = new CompanyCertificate(id, default, default, default, default, default);
+        initialize?.Invoke(entity);
+        _context.Attach(entity);
+        updateFields.Invoke(entity);
+    }
+
+    public void AttachAndModifyCompanyCertificateDocumentDetails(Guid id, Action<Document>? initialize, Action<Document> updateFields)
+    {
+        var entity = new Document(id, null!, null!, null!, default, default, default, default);
+        initialize?.Invoke(entity);
+        _context.Attach(entity);
+        updateFields.Invoke(entity);
+    }
+
+    public Task<(byte[] Content, string FileName, MediaTypeId MediaTypeId, bool Exists)> GetCompanyCertificateDocumentByCompanyIdDataAsync(Guid documentId, Guid companyId, DocumentTypeId documentTypeId) =>
+        _context.Documents
+        .Where(x => x.Id == documentId &&
+               x.DocumentTypeId == documentTypeId &&
+               x.CompanyUser!.Identity!.CompanyId == companyId)
+        .Select(x => new ValueTuple<byte[], string, MediaTypeId, bool>(x.DocumentContent, x.DocumentName, x.MediaTypeId, x.Id == documentId))
```

Comment:
The exists can always be set to true, no check needed because it's in the where clause.

```suggestion
        .Select(x => new ValueTuple<byte[], string, MediaTypeId, bool>(x.DocumentContent, x.DocumentName, x.MediaTypeId, true))
```

Reply:
Id comparison removed and added status as true